DOCS-8805 proxy config (#168)
mulesoft/docs-mule-sdk: modules/ROOT/pages/HTTP-based-connectors.adoc

        </oauth2:client-credentials-grant-type>
</http:request-config>
----
+

+
When configuring for the Authorization Code grant type, use the following snippet:
+

+
----
<http:request-config name="HTTP_Request_Configuration" host="api.github.com" protocol="HTTPS" port="443" usePersistentConnections="false" doc:name="HTTP Request Configuration">
        <oauth2:authorization-code-grant-type clientId="00a3d08a823faf378568" clientSecret="19c09306adbb84c2d4bc99c585df51d49fe858cf" redirectionUrl="http://localhost:8082/callback">;
            <oauth2:authorization-request authorizationUrl="https://github.com/login/oauth/authorize"" localAuthorizationUrl="http://localhost:8082/login"/>;
            <oauth2:token-request tokenUrl="https://github.com/login/oauth/access_token">;
                <oauth2:token-response accessToken="#[payload.'access_token']" refreshToken="#[payload['access_token']]"/>
            </oauth2:token-request>
            <spring:property name="proxyConfig" ref="proxyConfig"/>
        </oauth2:authorization-code-grant-type>
</http:request-config>
----
+

+
Note that there must be a `@Parameter` annotation in the connection provider with

> the proxyConfig type so the connection refers to the proxy, for example:

Reviewer: When using the `proxyConfig` type, the connection provider must include a `@Parameter` annotation so that the connection refers to the proxy, for example:

Reviewer: Trying to remove the passive voice (and the "note that" construction), but not sure if that suggestion is technically correct. The sentence is a little hard to parse.

+

+
[source,java,linenums]
----
@Parameter
@Optional
@Placement(tab = "Proxy", order = 1)
private MyConnectorProxyConfiguration proxyConfig;
----
+

+

> Also, note that the proxy configuration must be set in the HTTP Client:

Reviewer: ```suggestion
You must also set the proxy configuration in the HTTP Client:
```

+

+
[source,java,linenums]
----
.setProxyConfig(connectionParams.getProxyConfig())
----
+

+
For more information, refer to the https://github.com/mulesoft/mule-http-connector/blob/9914ce6aa6ceb3aa16a5b821289f15b9c6af6acc/src/main/java/org/mule/extension/http/api/request/proxy/DefaultProxyConfig.java[Java code]

> of how HTTP Connector handles the proxy.

Reviewer: ```suggestion
for how HTTP Connector handles the proxy.
```
